apps/eth-wrapper/src/components/Wrapper.tsx

```diff
@@ -0,0 +1,127 @@
+import { useSafeAppsSDK } from '@gnosis.pm/safe-apps-react-sdk';
+import { SafeAppsSdkProvider } from '@gnosis.pm/safe-apps-ethers-provider';
+import { Snackbar } from '@material-ui/core';
+import Grid from '@material-ui/core/Grid';
+import { Button, TextField } from '@gnosis.pm/safe-react-components';
+import { getWethAddress, Erc20 } from '../utils/Erc20Constants';
+import { ethers } from 'ethers';
+import { WETHwithdraw_function } from '../utils/WETHConstants';
+
+interface WrapperProps {
+    wrap: boolean
+}
+
+const Wrapper: React.FC<WrapperProps> = (props: WrapperProps) => {
+    const { sdk, safe } = useSafeAppsSDK();
+
+    const [amountToWrap, setAmountToWrap] = useState("");
+    const [errorMessage, setErrorMessage] = useState("");
+    const [availableBalance, setAvailableBalance] = useState(0.0);
+    const [isError, setIsError] = useState(false);
+    const [safeTxHash, setSafeTxHash] = useState("");
+
+    const provider = useMemo(() => new SafeAppsSdkProvider(safe, sdk), [safe, sdk]);
+    const weth = useMemo(() => new ethers.Contract(getWethAddress(safe.network.toLowerCase()), Erc20, provider), [provider, safe]);
+
+    const wrapEth = useCallback(async () => {
+        if (isError) {
+            return;
+        }
+
+        if (props.wrap) {
+            try {
+                const parsedAmount = ethers.utils.parseEther(amountToWrap)
+                const safeTx = await sdk.txs.send({
+                    txs: [{
+                        to: getWethAddress(safe.network.toLowerCase()),
+                        value: parsedAmount.toString(),
+                        data: '0x'
+                    }]
+                })
+                setSafeTxHash(safeTx.safeTxHash);
+                console.log(safeTx.safeTxHash);
+            } catch (e) {
+                console.error(e)
+            }
+        } else {
+            try {
+                const parsedAmount = ethers.utils.parseEther(amountToWrap);
+                const withdraw = new ethers.utils.Interface(WETHwithdraw_function);
+                const safeTx = await sdk.txs.send({
+                    txs: [{
+                        to: getWethAddress(safe.network.toLowerCase()),
+                        value: '0',
+                        data: withdraw.encodeFunctionData("withdraw", [parsedAmount])
+                    }]
+                })
+                setSafeTxHash(safeTx.safeTxHash);
+                console.log(safeTx.safeTxHash);
+            } catch (e) {
+                console.error(e);
+            }
+        }
+    }, [sdk, amountToWrap, isError, props.wrap, safe])
+
+    const validateAmout = useCallback((newValue: string) => {
+        console.log("available balance:", availableBalance);
+        console.log("new value: ", newValue)
+        if (isNaN(Number(newValue))) {
+            setIsError(true);
+            setErrorMessage("Not a number");
+        }
+        else if (Number.parseFloat(newValue) > availableBalance) {
```

Comment:
Here I would use BigNumber comparision instead of converting both to a float and comparing them